### PR TITLE
Email users who haven't logged in for a while (Issue #25)

### PR DESCRIPTION
**Pull request by steven-ferguson**
https://github.com/code-scouts/green_mercury/issues/25

### CODE REVIEW

File: lib/user.rb [hunk 1]
```diff
@@ -180,6 +204,16 @@ def organizer?(event)
       organizer.user_uuid == user_uuid
     end
   end
+
+  def update_attribute(attribute_hash)
```
Comment (drewchandler) [inline lib/user.rb:208]: This should probably be called `update_attributes`.

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -59,6 +59,11 @@ def associated_users(collection)
     User.fetch_from_uuids(user_uuids)
   end
 
+  def update_last_logged_in
+    return unless current_user
+    current_user.update_attribute(last_logged_in: "#{Time.now}")
```
Comment (drewchandler) [inline app/controllers/application_controller.rb:64]: Do you really need to convert the time to a string? I think httparty will take care of that for you. If you do, `.to_s` is the more idiomatic way to do the conversion.

File: app/controllers/session_controller.rb [hunk 3]
```diff
@@ -3,6 +3,8 @@
 class SessionController < ApplicationController
   skip_before_filter :new_applicant
   skip_before_filter :pending_applicant
+  before_filter :update_last_logged_in, only: :logout
+  after_filter :update_last_logged_in, only: :acquire_session
```
Comment (ErinCall) [inline app/controllers/session_controller.rb:7]: Sessions aren't set to expire, so I think we actually need to call update_last_logged_in on every request. I guess it's more "last visited" than "last logged in." The goal is basically to pull people back in if they haven't looked at the site in a while.

File: spec/controllers/application_controller_spec.rb [hunk 4]
```diff
@@ -94,4 +94,28 @@
...
+            attributes: {last_logged_in: "#{Time.now}"},
+            client_id: 'fakeclientidfortests',
+            client_secret: 'fakeclientsecretfortests'
+          }
+        }
+      )
+      controller.update_last_logged_in
+    end
+
+    it 'should return nil there is no user' do
```
Comment (ErinCall) [inline spec/controllers/application_controller_spec.rb:117]: This test description needs an "if"

File: spec/controllers/application_controller_spec.rb [hunk 5]
```diff
@@ -94,4 +94,28 @@
...
+      controller.update_last_logged_in.should be_nil
+    end
+  end
```
Comment (ErinCall) [inline spec/controllers/application_controller_spec.rb:120]: ~~There should be a test that demonstrates that users are only emailed periodically~~

Oh now I see that that test is farther down

**Comment by ErinCall** [comment]
@steven-ferguson this is great and I'm gonna steal it! I'm going to close this pull request and address the code review notes on my own fork :~)

**Pull request closed without merging**